src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/AdmissionControlAction.java

```diff
@@ -0,0 +1,231 @@
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.ADMISSION_CONTROL;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.AppContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.RcaConf;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.annotations.SerializedName;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class AdmissionControlAction extends SuppressibleAction {
+
+    public static final String NAME = "AdmissionControlAction";
+
+    protected static final long DEFAULT_COOL_OFF_PERIOD_IN_MILLIS = TimeUnit.MINUTES.toMillis(5);
+
+    private final NodeKey esNode;
+    private final String controllerName;
+    private final boolean canUpdate;
+    private final double desiredValue;
+    private final double currentValue;
+
+    public AdmissionControlAction(
+            AppContext appContext,
+            final NodeKey esNode,
+            final String controllerName,
+            final boolean canUpdate,
+            final double desiredValue,
+            final double currentValue) {
+        super(appContext);
+        this.esNode = esNode;
+        this.canUpdate = canUpdate;
+        this.desiredValue = desiredValue;
+        this.currentValue = currentValue;
+        this.controllerName = controllerName;
+    }
+
+    public static Builder newBuilder(
+            final NodeKey esNode,
+            final String controllerName,
+            final AppContext appContext,
+            final RcaConf conf) {
+        return new Builder(esNode, controllerName, appContext, conf);
+    }
+
+    public String getControllerName() {
+        return controllerName;
+    }
+
+    @Override
+    public boolean canUpdate() {
+        return this.canUpdate;
+    }
+
+    @Override
+    public long coolOffPeriodInMillis() {
+        return DEFAULT_COOL_OFF_PERIOD_IN_MILLIS;
+    }
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public List<NodeKey> impactedNodes() {
+        return Collections.singletonList(esNode);
+    }
+
+    @Override
+    public Map<NodeKey, ImpactVector> impact() {
+        // AdmissionControl tuning is bi-directional
+        // We mark increase/decrease pressure based on desired value
+        final ImpactVector impactVector = new ImpactVector();
+        if (desiredValue > currentValue) {
+            impactVector.increasesPressure(ADMISSION_CONTROL);
```

Comment:
Can we see indirect impact on HEAP because of this tuning. If yes, @khushbr Do you think we need to add them here?

Reply:
AdmissionControl would increase/decrease threshold based on available jvm, and will kick in only when node is under duress. For example, if JVM reaches 95% then the customer will see more 429 compared to when JVM was at 85%, eventually reducing node drops and 5xx. 
Increase in heap pressure will be driven from customer workload.

Reply:
Should you have a cool-off period when heap decreases to avoid flip-flops?

Reply:
Yes we have 5 mins cool-off period 
`protected static final long DEFAULT_COOL_OFF_PERIOD_IN_MILLIS = TimeUnit.MINUTES.toMillis(5);`

Reply:
I agree wtih Sruti. how are we going to define the impact of ADMISSION_CONTROL ? and what happen if let's say you have a different type of admission control action in the future and if they both have impact on ADMISSION_CONTROL, then one will be blocked by the other ?

Reply:
Each controller of admissioncontrol works in its own perview. When we add new controllers in admissioncontrol they would work on their own thresholds. Ofcourse these should be validated when we add more admissioncontrol AutoTune